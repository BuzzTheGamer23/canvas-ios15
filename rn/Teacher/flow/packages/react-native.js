// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
