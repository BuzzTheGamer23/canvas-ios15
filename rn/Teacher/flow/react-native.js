//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
