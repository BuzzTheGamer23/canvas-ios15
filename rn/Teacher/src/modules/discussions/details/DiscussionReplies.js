/* @flow */

import React, { Component } from 'react'
import {
  View,
  StyleSheet,
} from 'react-native'
import Reply from './Reply'
import Navigator from '../../../routing/Navigator'

export type Props = {
  reply: DiscussionReply,
  participants: { [key: string]: UserDisplay },
  navigator: Navigator,
}

export default class DiscussionReplies extends Component<any, Props, any> {
  render () {
    let participants = this.props.participants || {}
    let r = (<Reply participants={participants} reply={this.props.reply} depth={0} navigator={this.props.navigator}/>)

    return (
      <View style={style.container}>
        {r}
      </View>
    )
  }
}

const style = StyleSheet.create({
  container: { flex: 1 },
})
