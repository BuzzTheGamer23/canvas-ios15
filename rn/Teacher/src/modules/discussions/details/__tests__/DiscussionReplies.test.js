/* @flow */

import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import DiscussionReplies, { type Props } from '../DiscussionReplies'

jest.mock('Button', () => 'Button').mock('TouchableHighlight', () => 'TouchableHighlight').mock('TouchableOpacity', () => 'TouchableOpacity')

const template = {
  ...require('../../../../api/canvas-api/__templates__/discussion'),
  ...require('../../../../api/canvas-api/__templates__/users'),
  ...require('../../../../__templates__/helm'),
}

jest.mock('WebView', () => 'WebView')

describe('DiscussionReplies', () => {
  let props
  let user = template.userDisplay()
  beforeEach(() => {
    props = {
      reply: template.discussionReply(),
      depth: 0,
      participants: { [user.id]: user },
      courseID: '1',
      discussionID: '1',
      deleteDiscussionEntry: jest.fn(),
      replyToEntry: jest.fn(),
      navigator: template.navigator(),
      pathIndex: 0,
    }
  })

  it('renders', () => {
    testRender(props)
  })

  function testRender (props: Props) {
    expect(render(props).toJSON()).toMatchSnapshot()
  }

  function render (props: Props): any {
    return renderer.create(<DiscussionReplies {...props}/>)
  }
})
