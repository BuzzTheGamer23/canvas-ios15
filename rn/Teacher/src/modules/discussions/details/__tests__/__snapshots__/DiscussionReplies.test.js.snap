// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionReplies renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "padding": 0,
        }
      }
    >
      <View
        accessibilityLabel=""
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#C7CDD1",
              "borderWidth": 0.5,
              "justifyContent": "center",
              "marginRight": 16,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 12,
              "height": 24,
              "width": 24,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "fontWeight": "600",
              },
              Object {
                "fontSize": 8,
              },
            ]
          }
        >
          TD
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#f5f5f5",
            "flex": 1,
            "width": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "marginTop": 12.8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "fontSize": 14,
                "fontWeight": "600",
              },
              Object {
                "fontFamily": ".SFUIDisplay-semibold",
              },
            ]
          }
        >
          The Donald
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#A5AFB5",
                "fontSize": 12,
                "marginBottom": 16,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          May 23, 2017 11:12 AM
        </Text>
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": ".SFUIDisplay",
              },
              Object {
                "color": "#2D3B45",
                "fontSize": 16,
              },
              Object {
                "color": "#A5AFB5",
                "fontSize": 14,
                "marginTop": 16,
              },
              Object {
                "fontFamily": ".SFUIDisplay",
              },
            ]
          }
        >
          Reply | Edit
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
